pyartcd/pyartcd/pipelines/promote.py

```diff
@@ -375,6 +401,214 @@ def _reraise_if_not_permitted(self, err: VerificationError, code: str, permits:
+            with open(f"{client_mirror_dir}/opm-{platform}-{release_name}.tar.gz", 'rb') as f:  # calc shasum
+                shasum = hashlib.sha256(f.read()).hexdigest()
+            with open(f"{client_mirror_dir}/sha256sum.txt", 'a') as f:  # write shasum to sha256sum.txt
+                f.write(f"{shasum}  opm-{platform}-{release_name}.tar.gz\n")
+
+    async def publish_multi_client(self, working_dir, from_release_tag, release_name, arch_list, client_type):
+        # Anything under this directory will be sync'd to the mirror
+        base_to_mirror_dir = f"{working_dir}/to_mirror/openshift-v4"
+        shutil.rmtree(f"{base_to_mirror_dir}/multi", ignore_errors=True)
+        release_mirror_dir = f"{base_to_mirror_dir}/multi/clients/{client_type}/{release_name}"
+        current_path = os.getcwd()
+
+        for go_arch in [go_arch_for_brew_arch(arch) for arch in arch_list]:
+            if go_arch == "multi":
+                continue
+            # From the newly built release, extract the client tools into the workspace following the directory structure
+            # we expect to publish to mirror
+            client_mirror_dir = f"{release_mirror_dir}/{go_arch}"
+            os.makedirs(client_mirror_dir)
+            # extract release clients tools
+            extract_release_client_tools(f"{constants.QUAY_RELEASE_REPO_URL}:{from_release_tag}", f"--to={client_mirror_dir}", go_arch)
+            # create symlink for clients
+            self.create_symlink(path_to_dir=client_mirror_dir, log_tree=True, log_shasum=True)
+
+        # Create a master sha256sum.txt including the sha256sum.txt files from all subarches
+        # This is the file we will sign -- trust is transitive to the subarches
+        for dir in os.listdir(release_mirror_dir):
+            if not os.path.isdir(f"{release_mirror_dir}/{dir}"):
+                continue
+            for root, dirs, files in os.walk(f"{release_mirror_dir}/{dir}"):
+                if "sha256sum.txt" not in files:
+                    continue
+                with open(f"{root}/sha256sum.txt", "rb") as f:
+                    shasum = hashlib.sha256(f.read()).hexdigest()
+                with open(f"{release_mirror_dir}/sha256sum.txt", 'a') as f:  # write shasum to sha256sum.txt
+                    f.write(f"{shasum}  {dir}/sha256sum.txt\n")
+        util.log_dir_tree(release_mirror_dir)
+
+        # Publish the clients to our S3 bucket.
+        await exectools.cmd_assert_async(f"aws s3 sync --no-progress --exact-timestamps {base_to_mirror_dir}/multi s3://art-srv-enterprise/pub/openshift-v4/multi", stdout=sys.stderr)
+
+    def create_symlink(self, path_to_dir, log_tree, log_shasum):
+        # External consumers want a link they can rely on.. e.g. .../latest/openshift-client-linux.tgz .
+        # So whatever we extract, remove the version specific info and make a symlink with that name.
+        current_path = os.getcwd() # /mnt/workspace/jenkins/working/build_promote-assembly
```

Comment:
No need to get the current directory. Artifacts to work with should be in `path_to_dir`